CMSgov/cmcs-eregulations
# EREGCSC-1304 - Cache Explorer on prod + Vite and Vuetify

Resolves [EREGCSC-1304](https://jiraent.cms.gov/browse/EREGCSC-1304)

**Description**
To enable policy researchers to use elements of the prototypes that tested well, we need to update the production site to be compatible with those elements.

**This pull request changes:**

1. Adds [Vite](https://vitejs.dev/) to `solution/ui/regulations` project to bundle and transpile Vue components and views that are being promoted from the prototype to the pilot site.
   - the location for these promoted Vue files is in `solution/ui/regulations/eregs-vite`
   - Vite output is sent to `solution/static-assets/regulations/vite`
   - These Vite-generated static assets (currently css and js bundles) can then be accessed in Django templates.  See [solution/backend/regulations/templates/regulations/cache.html](https://github.com/CMSgov/cmcs-eregulations/pull/416/files#diff-fc50304a149062d4e7322b41965914c316ca5fed46cbcaac6cbc6f9d2a42aaf1).
   - Vite works alongside the existing Rollup build process - Rollup is still used for "legacy" Vue components; Vite will be used for Vue components promoted from the Prototype.
2. Configures Vite to work with Vue 2, Vuetify 2, and Vue Router 3.
3. Promotes `CacheExplorer.vue` to `solution/ui/regulations` and mounts it in the `cache.html` Django template and serves it at `<baseUrl>/cache/`.
4. Promotes cache helper api methods from prototype to pilot site in [solution/ui/regulations/js/api.js](https://github.com/CMSgov/cmcs-eregulations/pull/416/files#diff-9e223a33e0ae6f8dcbd2f7bd603643c4f38d8eceaf82091cb4f4826010ba57d2).
5. Creates stub route for resources at `<baseUrl>/resources/` to ensure that Vue Router is working properly.

**Steps to manually verify this change...**

Test Deploy experimental site
1. visit the [experimental deployment cache route](https://6rn22vynye.execute-api.us-east-1.amazonaws.com/dev-416/cache/). Notice that there are no cached API calls.
2. Start browsing the [experimental deployment](https://6rn22vynye.execute-api.us-east-1.amazonaws.com/dev-416) so that you're sure you've made a bunch of requests.
    - Example: visit [https://6rn22vynye.execute-api.us-east-1.amazonaws.com/dev-416/42/433/Subpart-A/2021-03-01/](https://6rn22vynye.execute-api.us-east-1.amazonaws.com/dev-416/42/433/Subpart-A/2021-03-01/) and see that Supplemental Content is loaded.  Browse around additional subparts to load more supplemental content.
    - Open the console and look for CACHE MISS the first time you visit a page, and CACHE HIT when you reload the page.
3. Return to the [experimental deployment cache route](https://6rn22vynye.execute-api.us-east-1.amazonaws.com/dev-416/cache/). There should now be a bunch of API calls listed.
4. Delete all cache items.
5. Return to the Subparts where you were previously browsing and look for CACHE MISS in the dev console.

Test that everything builds locally
1. in `solution`, run `make local` to start the local docker environment, build all static assets (Vite and Rollup), and load Title 42.
    - run `make local.seed` to seed local DB with supplemental content.
2. run `make watch-all` to watch changes to JS and CSS (Rollup and Vite).
3. Open your IDE and make edits to any of the promoted `eregs-vite` js/css files.  
4. Refresh the local site to see your new changes.
